docs/root/overview.md

```diff
@@ -1,60 +1,75 @@
 # Nighthawk: architecture and key concepts
 
-## High level interaction model
-
-**Process** creates one or more **Workers**. **Worker** will run **Sequencer**,
-which in turn queries **RateLimiter** for [request-release
-timings](terminology.md#request-release-timings). When it is time to release a
-request, **BenchmarkClient** will be requested to do so by **Sequencer**.
-**BenchMarkClient** will then ask its underlying **Pool** to create a
-**StreamDecoder** for releasing the actual request. **StreamDecoder** will query the
-request data it needs to send from the configured **RequestSource**, and send it
-off. **StreamDecoder** will emit events as it progresses (pool ready,
-completion, etc), and timings will subsequently be recorded into **Statistic**
-as well as get bubbled up to **Sequencer** for tracking in-flight work and
-**Statistic** bookkeeping.
-
-**Sequencer** will query the configured **TerminationPredicates** to terminate
-when and how to terminate execution. When all **Workers** have finished,
-**Process** will collect the results from all **Workers** via
-**OutputCollector**, and run **OutputFormatter** to transform to the requested
-output format.
-
-However, it is said that pictures say more then words, so here is a block
-diagram.
-
-![Draft diagram](draft-high-level-diagram.png)
-
-[TODO(#249):add a real diagram](https://github.com/envoyproxy/nighthawk/issues/249)
-
-## Notable upcoming changes
-
-Calling out two new concepts that may get proposed in the future, and cause some
-churn in the code base as we inject them.
-
-### Phases
-
-One notable addition / change that may get proposed in the near future is the
-introduction of **Phase**. **Phase** would represent a distinct stage of an
-execution, for example a warm-up. It would then be useful to have
-per-phase reporting of latencies as well as counters and latencies.
-
-Concretely, a warm-up phase could be represented by a distinct duration
-termination predicate and a ramping rate limiter. Upon completion, the `hot`
-BenchmarkClient with its associated pool would then be live-transferred to the
-next configured phase, after which execution can continue.
-
-One other reason to have this is that it would enable remote- and/or cli-
-controlled ramping of certain test parameters by associating those to different
-phases. Termination predicates can be leveraged to immediately terminate the
-current phase after injecting a new one, allowing for real-time steering via
-gRPC and/or CLI.
-
-### Streaming parameterization and output stats
-
-Once we have phases, the gRPC service, and perhaps the CLI, would be natural
-candidates to follow up with to allow dynamic phase injection, as well as send
-back reports per phase.
+## Architecture
+
+The following diagram outlines the overall architecture of Nighthawk.
+
+![Nighthawk architecture](images/nighthawk_architecture.png)
```

Comment:
I'm running github with the dark theme; the arrows are invisible in that case as they merge with the background.. perhaps use a white background instead of a transparant one? 

![image](https://user-images.githubusercontent.com/2239556/144191095-7b7a0a73-3a54-410a-bf73-b9bba3e91811.png)

Reply:
transparancy aside, do we want to show that:
- we can test multiple endpoints? 
- we can stream stats optionally to a separate backend during execution?

Reply:
Added white background in 188c83057665ec6db290f9fbb0f9750b6eee3db1.

As for the other two questions - I would lean towards not doing that in these diagrams. The goal here isn't to outline all the features Nighthawk has, but to help understand its overall architecture. The more details we add here, the more distracted we get from that goal. These features can be found in the command line options, or alternatively we could add a separate document listing Nighthawk's features.

Choosing not to address thee here, but please let me know if you feel strongly.